src/hooks/cgroups/pbs_cgroups.PY

```diff
@@ -2330,6 +2476,168 @@ class CgroupUtils(object):
                                 prefix + cgfile)
         return os.path.join(subdir, prefix + cgfile)
 
+    def morph_config_dict_bools(self, config_dict):
+        # function to morph "vntype in: vntype1,vntype2,..."
+        # and "vntype not in: ..." values
+        # into booleans depending on the vntype in the CgroupConfig object
+        # Ditto for "host in:" and "host not in:"
+        # Convert old run_only_on_hosts and exclude_vntypes to derive
+        # "enabled" boolean (which will now _always_ be defined)
+        pbs.logmsg(pbs.EVENT_DEBUG4, "%s: Config file parsing" % caller_name())
+        vntype = self.vntype
+        # avoid crashes in fnmatch if no vntype was specified on the host
+        if not vntype:
+            vntype = "__UNKNOWN__"
+        for key_found, value_found in config_dict.iteritems():
+            if isinstance(value_found, dict):
+                # subsection found
+                pbs.logmsg(pbs.EVENT_DEBUG4,
+                           "%s: Cfg file parsing, subsection for %s"
+                           % (caller_name(), key_found))
+                self.morph_config_dict_bools(value_found)
+                pbs.logmsg(pbs.EVENT_DEBUG4,
+                           "%s: Cfg file parsing, finished subsection for %s"
+                           % (caller_name(), key_found))
+            elif isinstance(value_found, basestring):
+                # string value -- could be morphable description
+                value_split = value_found.strip().split(':', 1)
+                if ((value_split[0].lower().strip() == 'vntype in')
+                        and (len(value_split) > 1)):
```

Comment:
The order of these two conditions need to be switched. The length of `value_split` should be checked first before accessing `value_split[0]`.

Reply:
Correct.